packages/firebase_ui_auth/lib/src/widgets/password_input.dart

```diff
@@ -44,21 +51,48 @@ class PasswordInput extends StatelessWidget {
     required this.placeholder,
     this.autofillHints = const [AutofillHints.password],
     this.validator,
+    this.showVisibilityToggle = false,
   });
 
+  @override
+  State<PasswordInput> createState() => _PasswordInputState();
+}
+
+class _PasswordInputState extends State<PasswordInput> {
+  var obscureText = true;
+
+  Widget buildSuffixIcon() {
+    final mIcon = obscureText ? Icons.visibility : Icons.visibility_off;
+    final cIcon = obscureText ? CupertinoIcons.eye : CupertinoIcons.eye_slash;
+
+    return UniversalIconButton(
+      materialIcon: mIcon,
+      cupertinoIcon: cIcon,
+      onPressed: () => setState(() => obscureText = !obscureText),
+    );
+  }
```

Comment:
Should we allow a custom icon?

Reply:
the whole library tries to follow material and Cupertino spec, so I'd say no.

Reply:
could be implemented later if we ever get a feature request for it